repo/packages/B/bookkeeper/1/config.json

```diff
@@ -1,52 +1,95 @@
-                },
-                "zk_client": {
-                    "default": "master.mesos:2181",
-                    "description": "env -- zookeeper client instance",
-                    "type": "string"
-                },
-                "service_port": {
-                    "default": 3181,
-                    "description": "bookkeeper export service port, using PORT0 in marathon",
-                    "type": "integer"
-                }
-            },
-            "required": ["cpus", "mem", "instances"],
-            "type": "object"
+  "$schema": "http://json-schema.org/schema#",
+  "properties": {
+    "bookkeeper": {
+      "description": "bookkeeper specific configuration properties",
+      "properties": {
+        "name": {
+          "default": "bookkeeper",
+          "description": "The name of this DC/OS service.",
+          "type": "string"
+        },
+        "cpus": {
+          "default": 1,
+          "description": "CPU shares to allocate to each bookkeeper instance.",
+          "minimum": 0.5,
+          "type": "number"
+        },
+        "instances": {
+          "default": 1,
+          "description": "Number of bookkeeper instances to run.",
+          "minimum": 1,
+          "type": "integer"
+        },
+        "mem": {
+          "default": 1024.0,
+          "description": "Memory (MB) to allocate to each bookkeeper task.",
+          "minimum": 512.0,
+          "type": "number"
+        },
+
+        "ledgers_volume_size": {
+          "default": 20000,
+          "description": "persistent Volume size(MB) for bookkeeper ledgers data.",
+          "minimum": 10000,
+          "type": "number"
+        },
+        "index_volume_size": {
+          "default": 7000,
+          "description": "persistent Volume size(MB) for bookkeeper index data.",
+          "minimum": 2000,
+          "type": "number"
+        },
+        "journal_volume_size": {
+          "default": 4000,
+          "description": "persistent Volume size(MB) for bookkeeper journal data.",
+          "minimum": 2000,
+          "type": "number"
+        },
+
+        "BK_BOOKIE_EXTRA_OPTS": {
+          "default": "\"-Xms1g -Xmx1g -XX:MaxDirectMemorySize=1g -XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB\"",
+          "description": "env -- bookkeeper JVM runtime options",
+          "type": "string"
+        },
+        "BK_zkServers": {
+          "default": "master.mesos:2181",
+          "description": "env -- zookeeper server instance",
+          "type": "string"
+        },
+        "BK_bookiePort": {
+          "default": 3181,
+          "description": "env -- bookkeeper export service port, using PORT0 in marathon",
+          "type": "integer"
+        },
+        "BK_journalDirectory": {
```

Comment:
these directories all look like required parameters to me

Reply:
Thanks. All these vars should be required. will change it.